vim-engine/src/main/kotlin/com/maddyhome/idea/vim/common/ExtendedPictographics.kt

```diff
@@ -0,0 +1,81 @@
+package com.maddyhome.idea.vim.common
+
+/**
+ * Answers whether a given code point is a unicode Extended_Pictographic.
+ *
+ * NOTE: this is a part of the Java 21 API. Can be removed once we start targeting that version.
+ */
+internal fun isExtendedPictographic(codePoint: Int): Boolean {
+  // Outside of the bitmap.
+  if (codePoint >= bitmap.size * 64) return false
+
+  val idx = codePoint / 64
+  val bit = codePoint % 64
+  val bucket = bitmap[idx]
+
+  return (bucket and (1L shl bit)) != 0L
+}
+
+// A bitmap that maps a code point into whether it has the Extended_Pictographic property.
+// The code points go in increasing order by index and in reverse order by bit in a specific long.
+// This way a simple divmod is enough to compute both indices.
+private val bitmap = longArrayOf(
```

Comment:
Where this bitmap comes from?

Reply:
I've generated it with a script from emoji-data.txt from the unicode character database. I could add the source file into the repository and add some code that would generate this bitmap on demand, is there any infrastructure of that in idea-vim or what would be the preferred way to do this?

Reply:
This won't be needed with JDK 21, right? If so, it's okay to just not where this map comes from so the people didn't wounder about these magic numbers.

Reply:
Yes, JDK21 contains this functionality. Very well, I'll add a link to the file in the comment then.